deeplay/components/rnn.py

```diff
@@ -62,65 +61,142 @@ def normalization(self) -> LayerList[Layer]:
         self._in_features = in_features
         self.hidden_features = hidden_features
         self.out_features = out_features
-        self.num_layers = num_layers
         self.rnn_type = rnn_type
         self.bidirectional = bidirectional
         self.batch_first = batch_first
 
-            
+        if in_features is None:
+            raise ValueError("in_features must be specified")
+        if hidden_features is None or hidden_features == []:
+            raise ValueError("hidden_features must be specified")
         if in_features is not None and in_features <= 0:
             raise ValueError(f"in_channels must be positive, got {in_features}")
 
-        if hidden_features <= 0:
+        if any(h <= 0 for h in hidden_features):
             raise ValueError(
                 f"all hidden_channels must be positive, got {hidden_features}"
             )
-
-        self.blocks = LayerList()
-        rnn_class = torch.nn.LSTM if self.rnn_type == 'LSTM' else torch.nn.GRU
-        self.rnn = rnn_class(
-            in_features,
-            hidden_features,
-            num_layers,
-            dropout=dropout,
-        )
-        #self.blocks.append(Layer(self.rnn))
-        self.blocks.append(self.rnn)
+        if out_features is not None:
+            if out_features <= 0:
+                raise ValueError(
+                    f"Number of output features must be positive, got {out_features}"
+                )
+        if rnn_type not in ["RNN", "LSTM", "GRU"]:
+            raise ValueError(
+                f"rnn_type must be one of 'RNN', 'LSTM', or 'GRU', got {rnn_type}"
+            )
         
-        if bidirectional:
-                layer_output_size = hidden_features * 2
+        if embedding:
+            self.embedding_dropout=torch.nn.Dropout(dropout)
+
+        if self.rnn_type == 'LSTM':
+             self.rnn_class = torch.nn.LSTM
+        elif self.rnn_type == 'GRU':
+            self.rnn_class = torch.nn.GRU
+        else:
+            self.rnn_class = torch.nn.RNN
+             
+        if isinstance(out_activation, torch.nn.Module):
+            self.out_activation=Layer(out_activation)
+            print(self.out_activation)
         else:
-            layer_output_size = hidden_features
+            if out_activation == 'softmax':
+                self.out_activation = Layer(torch.nn.Softmax,dim=1)
+            elif out_activation == 'sigmoid':
+                self.out_activation = Layer(torch.nn.Sigmoid)
+            elif out_activation == 'tanh':
+                self.out_activation = Layer(torch.nn.Tanh)
+            elif out_activation == 'relu':
+                self.out_activation = Layer(torch.nn.ReLU)
+            elif out_activation == 'leaky_relu':
+                self.out_activation = Layer(torch.nn.LeakyReLU)
+            elif out_activation == 'gelu':
+                self.out_activation = Layer(torch.nn.GELU)
+            else:
+                self.out_activation = Layer(torch.nn.Identity)
```

Comment:
While I support the idea of allowing users to pick an activation using a string, it should be abstracted away and defined outside the class as a registry that any model can access. So we are consistent in mlp and cnn